gguf-py/gguf/constants.py

```diff
@@ -33,21 +33,22 @@ class General:
         FILE_TYPE            = "general.file_type"
 
     class LLM:
-        VOCAB_SIZE                 = "{arch}.vocab_size"
-        CONTEXT_LENGTH             = "{arch}.context_length"
-        EMBEDDING_LENGTH           = "{arch}.embedding_length"
-        BLOCK_COUNT                = "{arch}.block_count"
-        LEADING_DENSE_BLOCK_COUNT  = "{arch}.leading_dense_block_count"
-        FEED_FORWARD_LENGTH        = "{arch}.feed_forward_length"
-        EXPERT_FEED_FORWARD_LENGTH = "{arch}.expert_feed_forward_length"
-        USE_PARALLEL_RESIDUAL      = "{arch}.use_parallel_residual"
-        TENSOR_DATA_LAYOUT         = "{arch}.tensor_data_layout"
-        EXPERT_COUNT               = "{arch}.expert_count"
-        EXPERT_USED_COUNT          = "{arch}.expert_used_count"
-        EXPERT_SHARED_COUNT        = "{arch}.expert_shared_count"
-        EXPERT_WEIGHTS_SCALE       = "{arch}.expert_weights_scale"
-        POOLING_TYPE               = "{arch}.pooling_type"
-        LOGIT_SCALE                = "{arch}.logit_scale"
+        VOCAB_SIZE                        = "{arch}.vocab_size"
+        CONTEXT_LENGTH                    = "{arch}.context_length"
+        EMBEDDING_LENGTH                  = "{arch}.embedding_length"
+        BLOCK_COUNT                       = "{arch}.block_count"
+        LEADING_DENSE_BLOCK_COUNT         = "{arch}.leading_dense_block_count"
+        FEED_FORWARD_LENGTH               = "{arch}.feed_forward_length"
+        EXPERT_FEED_FORWARD_LENGTH        = "{arch}.expert_feed_forward_length"
+        SHARED_EXPERT_FEED_FORWARD_LENGTH = "{arch}.shared_expert_feed_forward_length"
```

Comment:
I wonder, (since this is a new key and can still be renamed) should this be `EXPERT_SHARED_FEED_FORWARD_LENGTH` and `{arch}.expert_shared_feed_forward_length` to match the naming of `EXPERT_SHARED_COUNT`?

This way, anything related to experts would still start with "expert_".

But maybe this is fine as-is too. What do you think?

(Side note: `EXPERT_SHARED_COUNT` seems to be used for a similar purpose in DeepSeekV2. In the case of Qwen2-57B-A14B, I think this could be `8` (`20480/2560 = 8`), but the shared expert feed-forward length seems more general (which is good))

Reply:
> I wonder, (since this is a new key and can still be renamed) should this be `EXPERT_SHARED_FEED_FORWARD_LENGTH` and `{arch}.expert_shared_feed_forward_length` to match the naming of `EXPERT_SHARED_COUNT`?
> 
> This way, anything related to experts would still start with "expert_".
> 
> But maybe this is fine as-is too. What do you think?

I have no preference for the naming of the key. @ggerganov just let me know and I will force-push the update with the changed key name `SHARED_EXPERT_FEED_FORWARD_LENGTH` -> `EXPERT_SHARED_FEED_FORWARD_LENGTH`

Reply:
Yes `EXPERT_SHARED_FEED_FORWARD_LENGTH` fits better into the naming convention because it reuses an existing prefix

Reply:
fixed in https://github.com/ggerganov/llama.cpp/compare/7fbb7a1b9dcdf57096d0d3e4fc517291ac6f8a23..d9452267a0069a2873f4855cfc0b671b31704889